Sources/SimpleCurrencyConverter/SimpleCurrencyConverter.swift

```diff
@@ -1,6 +1,140 @@
+        return container.resolve(type: CurrencyConverterProtocol.self)!
+    }
+    
+    private init(exchangeRatesAPIKey: String? = nil, urlSession: URLSession? = nil) {
+        if let exchangeRatesAPIKey = exchangeRatesAPIKey {
+            self.exchangeRatesAPIKey = exchangeRatesAPIKey
+        } else {
+            self.exchangeRatesAPIKey = apiKeyFromPlistFile
+        }
+
+        if let urlSession = urlSession {
+            self.networkManager = ExchangeRatesNetworkManager(session: urlSession)
+        } else {
+            self.networkManager = ExchangeRatesNetworkManager(session: URLSession.shared)
+        }
     }
+    
+    public func getExchangeRate(
+        baseCurrency: Currency,
+        targetCurrencies: [Currency],
+        completion: @escaping (Result<[Currency : Double], Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestExchangeRate(
+                    apiKey: exchangeRatesAPIKey, url: exchangeRatesURL.get(), base: baseCurrency, target: targetCurrencies)
+                switch result {
+                case .success(let rates):
+                    completion(.success(rates))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
+                }
+            }
+        }
+    
+    public func convert(
+        amount: Double,
+        baseCurrency: Currency,
+        targetCurrency: Currency,
+        completion: @escaping (Result<Double, Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestConvert(
+                    apiKey: exchangeRatesAPIKey, url: convertURL.get(), amount: amount, base: baseCurrency, target: targetCurrency)
+                switch result {
+                case .success(let result):
+                    completion(.success(result))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
+                }
+            }
+        }
+    
+    public func convert(
+        amount: Double,
+        baseCurrency: Currency,
+        targetCurrencies: [Currency],
+        completion: @escaping (Result<[Currency : Double], Error>) -> Void) {
+            Task {
+                let result = try await networkManager?.requestExchangeRate(
+                    apiKey: exchangeRatesAPIKey, url: exchangeRatesURL.get(), base: baseCurrency, target: targetCurrencies)
+                switch result {
+                case .success(let rates):
+                    var modifiedRates = rates
+                    for rate in modifiedRates {
+                        let value = rate.value * amount
+                        modifiedRates.updateValue(value, forKey: rate.key)
+                    }
+                    completion(.success(modifiedRates))
+                case .failure(let error):
+                    completion(.failure(error))
+                default: ()
```

Comment:
@ts-dzmitry-kurski Does it build without error if you remove `default: ()`?

Reply:
No, `Switch must be exhaustive`. So default or .none case should be implemented.

Reply:
ah it's because `result` is an Optional type. Actually this case should be handled as well. Otherwise you risk having unfulfilled promise issue. You could just return `completion(.failure(Error))` here or maybe it's better to unwrap `networkManager` in a `guard let` and then have switch without `default:` case